extensions/tokenizers/src/main/java/ai/djl/huggingface/tokenizers/HuggingFaceTokenizer.java

```diff
@@ -305,60 +306,98 @@ public String decode(long[] ids) {
     /**
      * Sets the truncation and padding behavior for the tokenizer.
      *
-     * @param truncationStrategy the {@code TruncationStrategy} to use
-     * @param paddingStrategy the {@code PaddingStrategy} to use
-     * @param maxLength the maximum length to pad/truncate sequences to
+     * @param truncation whether to enable truncation
+     * @param padding whether to enable padding
+     * @param maxLength maximum length of sequences to truncate and/or pad to
      */
-    public void setTruncationAndPadding(
-            TruncationStrategy truncationStrategy,
-            PaddingStrategy paddingStrategy,
-            long maxLength) {
-        setTruncationAndPadding(truncationStrategy, paddingStrategy, maxLength, 0, 0);
+    public void setTruncationAndPadding(boolean truncation, boolean padding, long maxLength) {
+        setTruncationAndPadding(Boolean.toString(truncation), Boolean.toString(padding), maxLength);
     }
 
     /**
      * Sets the truncation and padding behavior for the tokenizer.
      *
-     * @param truncationStrategy the {@code TruncationStrategy} to use
-     * @param paddingStrategy the {@code PaddingStrategy} to use
-     * @param maxLength the maximum length to pad/truncate sequences to
-     * @param stride value to use when handling overflow
+     * @param truncation the type of truncation behavior to use
+     * @param padding the type of padding behavior to use
+     * @param maxLength maximum length of sequences to truncate and/or pad to
      */
-    public void setTruncationAndPadding(
-            TruncationStrategy truncationStrategy,
-            PaddingStrategy paddingStrategy,
-            long maxLength,
-            long stride) {
-        setTruncationAndPadding(truncationStrategy, paddingStrategy, maxLength, stride, 0);
+    public void setTruncationAndPadding(String truncation, String padding, long maxLength) {
```

Comment:
Is there a reason for the intermediate string format for truncation here? Right now, it goes from `Boolean -> String -> TruncationStrategy`. I recommend picking one "standardized" format which is probably the typed `TruncationStrategy`. Then, if you have alternative strategies (boolean, maybe String) then it only goes from the alternative to the standardized format.

Beyond that, I would personally lean towards supporting the enum over a String format. It is easier and more reliable to use from Java code and easier to document. Speaking of which, it may be worth adding some javadoc comments to the enum to explain the different truncation and padding strategies

Reply:
Yea this is something I'm debating as well - prior to this we were leaning on the enum option where users would pass in an enum value to indicate the behavior. The overall motivation for this change is to adhere more to the fast tokenizer interface, which lets users pass in a boolean, String, or enum https://huggingface.co/transformers/v4.4.2/main_classes/tokenizer.html. 

On our side, I think it makes sense to support the boolean parameter, and then one of either enum or string. I'd prefer enum for the same reasons you pointed out. 

The two options I see are to expose `boolean and enum` methods, or `boolean and string` methods, of which i prefer the former.